src/Microsoft.AspNet.FileProviders.Combined/CombinedFileProvider.cs

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNet.FileProviders
+{
+    /// <summary>
+    /// Looks up files using embedded resources in the specified assembly.
+    /// This file provider is case sensitive.
+    /// </summary>
+    public class CombinedFileProvider : IFileProvider
+    {
+        private readonly IFileProvider[] _fileProviders;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CombinedFileProvider" /> class using a list of file provider.
+        /// </summary>
+        /// <param name="fileProviders"></param>
+        public CombinedFileProvider(params IFileProvider[] fileProviders)
+        {
+            _fileProviders = fileProviders;
+        }
+
+        /// <summary>
+        /// Locates a file at the given path.
+        /// </summary>
+        /// <param name="subpath">The path that identifies the file. </param>
+        /// <returns>The file information. Caller must check Exists property. This will be the first existing <see cref="IFileInfo"/> returned by the provided <see cref="IFileProvider"/> or a not found <see cref="IFileInfo"/> if no existing files is found.</returns>
+        public IFileInfo GetFileInfo(string subpath)
+        {
+            foreach (var fileProvider in _fileProviders)
+            {
+                var fileInfo = fileProvider.GetFileInfo(subpath);
+                if (fileInfo != null && fileInfo.Exists)
+                {
+                    return fileInfo;
+                }
+            }
+            return new NotFoundFileInfo(subpath);
+        }
+
+        /// <summary>
+        /// Enumerate a directory at the given path, if any.
+        /// </summary>
+        /// <param name="subpath">The path that identifies the directory</param>
+        /// <returns>Contents of the directory. Caller must check Exists property.
+        /// The content is a merge of the contents of the provided <see cref="IFileProvider"/>.
+        /// When there is multiple <see cref="IFileInfo"/> with the same Name property, only the first one is included on the results.</returns>
+        public IDirectoryContents GetDirectoryContents(string subpath)
+        {
+            // gets the content of all directories and merged them
+            var existingDirectoryContents = new List<IDirectoryContents>();
```

Comment:
Instead of enumerating up front it might make sense to have the `CombinedDirectoryContents` take both file providers and do this in a more lazy matter (possible using yield or a custom iterator).